presto-main/src/test/java/com/facebook/presto/block/BlockAssertions.java

```diff
@@ -708,15 +713,18 @@ public static DictionaryBlock createRandomDictionaryBlock(Block dictionary, int
 
     public static DictionaryBlock createRandomDictionaryBlock(Block dictionary, int positionCount, boolean isView)
     {
+        checkArgument(dictionary.getPositionCount() > 0, format("dictionary's positionCount %d is less than or equal to 0", dictionary.getPositionCount()));
+
         int idsOffset = 0;
         if (isView) {
             idsOffset = min(ThreadLocalRandom.current().nextInt(dictionary.getPositionCount()), 1);
         }
 
-        int[] ids = IntStream.range(0, positionCount + idsOffset).map(i -> ThreadLocalRandom.current().nextInt(dictionary.getPositionCount() / 10)).toArray();
+        int[] ids = IntStream.range(0, positionCount + idsOffset).map(i -> ThreadLocalRandom.current().nextInt(max(dictionary.getPositionCount() / 10, 1))).toArray();
         return new DictionaryBlock(idsOffset, positionCount, dictionary, ids, false, randomDictionaryId());
     }
 
+    // It is recommended to have positionCount >= (10 / nestedNullRate) when nestedNullRate > 0, so that the nullRate can be close to the desired value.
```

Comment:
@yingsu00 should this comment be updated to match the rest?

Reply:
@mbasmanova   Sorry missed this one. I have updated it to be
```
// Note: Make sure positionCount is sufficiently large if nestedNullRate or primitiveNullRate is greater than 0
public static Block createRandomBlockForType(
```